### ISSUE TITLE
Pianoteq presets cache is not properly invalidated

### ISSUE THREAD

**Issue by BurNiinTRee** [zynthian/zynthian-issue-tracking#681]
**Describe the bug**
After uploading a newer version of Pianoteq, newly added Instruments and Presets don't show up.

**To Reproduce**
1. Load the default Pianoteq 6 Trial version
2. Upload a new version of Pianoteq (in my case 7.5.4) to the zynthian through the webconfig 
3. (Add a license key)
4. Try to play one of the new instruments (e.g. Petrof Mistral, NY/HH Steinway D)
5. Not find it in the preset list

**Expected behavior**
Licensed, new instruments should appear under the `*** Favorites***` section, unlicensed new instruments should appear under the `--- DEMO Instruments` Section in the Pianoteq presets list.

**Screenshots**

**Setup:**
 - SD-image
 - Hardware Config
 - Software Version
```
 ##########################       ----------------------------------------
#                          #      ZynthianOS: Built on os.zynthian.org
#            ##            #      Timestamp: 2021-09-25
#      ##   #  #   ##      #      Optimized: Raspberry Pi 3 Model B Plus
#     #  #  #  #  #  #     #      ----------------------------------------
###   #  #  #  #  #  #   ###      Kit:           V4
#  #  #  #  #  #  #  #  #  #      Display:       ZynScreen 3.5 (v1)
#   ##   #  #  #  #   ##   #      Soundcard:     HifiBerry DAC+ ADC PRO
#         ##    ##         #      Wiring Layout: MCP23017_ZynScreen
#                          #      ----------------------------------------
 ##########################
                                  zynthian-ui: stable (30294c)
                                  zynthian-webconf: stable (796155)
                                  zyncoder: stable (7c0973)
                                  zynthian-sys: stable (0bd40a)
                                  zynthian-data: stable (82571e)
                                  ---------------------------------------
```

**Additional context**
This Issue persisted across reboots.
As a workaround, I deleted the `/zynthian/config/pianoteq6/presets_cache.json` file; this fixed the issue after a reboot.

**Comment by riban-bw** [comment]
@BurNiinTRee are you using the native version of Pianoteq (near top of list) or the LV2 version (enabled by webconf and appears in list of LV2 Instruments further down list)?

**Comment by BurNiinTRee** [comment]
Yes, I am talking about the native version, I haven't tried out the LV2 version yet.

**Comment by riban-bw** [comment]
Favourite is not for licensed instruments it is for any preset that you have manually tagged as a favourite.

I will try an update during testing this week to see if I can reproduce this issue.

**Comment by BurNiinTRee** [comment]
I hadn't yet figured out how the favorites section worked, what I meant was the list of the licensed instruments, located right between the `*** Favorites ***` option and the demo instruments.

**Comment by riban-bw** [comment]
Understood! Do you see any licensed instruments, e.g. free or ones you previously owned or are there no entries between the**favs** and **demo**?

Have you run the scan for new instruments and presets in webconf? Beware this is flawed and may bomb out in webconf but should eventually complete in the background. Give it 10 minutes then restart any retest.

**Comment by BurNiinTRee** [comment]
Yes, I saw the other licensed Instruments I owned, just not the Steinway Model D which was added in version 7.
I worked around it by deleting the presets cache (`/zynthian/config/pianoteq6/presets_cache.json`).
Tbh I did not see the scan so maybe it was user error after all. Do you mean the `Update Presets` button  below the `Upload Binary/PTQ` button?

**Comment by riban-bw** [comment]
Yes that will be the one!

**Comment by riban-bw** [comment]
I removed the configuration file that includes the license then re-licensed Pianoteq on the device. Only the _free_ (included with purchase) instruments showed with the rest below the "Demo Instruments" divider.

Pressing "Update Presets" button in webconf did not fix this, even after a reboot. It looks like the cache file is created in /zynthian/config/pianoteq6/presets_cache.json. I will look at the code...

**Comment by riban-bw** [comment]
Okay - this is now working for me:

- Use webconf to upload binary (Pianoteq software downloaded from Modartt user area)
- Use webconf to activate Pianoteq
- Use webconf to upload PTQ (instruments downloaded from Modartt user area)
- Use webconf to reactivate Pianoteq (maybe this could be done once - it was the step that made it work today)
- Use webconf to Update Presets

This is certainly arduous and could be optimised but it does work. @BurNiinTRee please confirm this works for you so that we can close this ticket. Feel free to raise another feature request to simplify the process.

**Comment by riban-bw** [comment]
No response from @BurNiinTRee for several years.
